src/main/java/com/commercetools/sync/sdk2/types/utils/FieldDefinitionUpdateActionUtils.java

```diff
@@ -0,0 +1,197 @@
+package com.commercetools.sync.sdk2.types.utils;
+
+import static com.commercetools.sync.sdk2.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.sdk2.commons.utils.OptionalUtils.filterEmptyOptionals;
+import static com.commercetools.sync.sdk2.types.utils.LocalizedEnumValueUpdateActionUtils.buildLocalizedEnumValuesUpdateActions;
+import static com.commercetools.sync.sdk2.types.utils.PlainEnumValueUpdateActionUtils.buildEnumValuesUpdateActions;
+
+import com.commercetools.api.models.type.CustomFieldEnumType;
+import com.commercetools.api.models.type.CustomFieldLocalizedEnumType;
+import com.commercetools.api.models.type.CustomFieldSetType;
+import com.commercetools.api.models.type.FieldDefinition;
+import com.commercetools.api.models.type.FieldType;
+import com.commercetools.api.models.type.TypeChangeLabelActionBuilder;
+import com.commercetools.api.models.type.TypeUpdateAction;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+
+/** This class is only meant for the internal use of the commercetools-sync-java library. */
+final class FieldDefinitionUpdateActionUtils {
+
+  /**
+   * Compares all the fields of an old {@link FieldDefinition} with a new {@link FieldDefinition}
+   * and returns a list of {@link TypeUpdateAction} as a result. If both the {@link FieldDefinition}
+   * and the {@link FieldDefinition} have identical fields, then no update action is needed and
+   * hence an empty {@link java.util.List} is returned.
+   *
+   * @param oldFieldDefinition the old field definition which should be updated.
+   * @param newFieldDefinition the new field definition where we get the new fields.
+   * @return A list with the update actions or an empty list if the field definition fields are
+   *     identical.
+   * @throws com.commercetools.sync.sdk2.commons.exceptions.DuplicateKeyException in case there are
+   *     localized enum values with duplicate keys.
+   */
+  @Nonnull
+  static List<TypeUpdateAction> buildActions(
+      @Nonnull final FieldDefinition oldFieldDefinition,
+      @Nonnull final FieldDefinition newFieldDefinition) {
+
+    final List<TypeUpdateAction> updateActions =
+        filterEmptyOptionals(buildChangeLabelUpdateAction(oldFieldDefinition, newFieldDefinition));
```

Comment:
i wonder why the changeOrderUpdateAction isn't built here... confusing.

Reply:
`ChangeEnumValueOrder` is actually [being called from this class](https://github.com/commercetools/commercetools-sync-java/pull/1007/files#diff-f95f200bf9c5f33efe8ace2227433aee08fb8dfcf597cc42a9b0e73f859454c2R105). I guess it's not there because the update is for enum type and its values only, so it's not in the field definition itself.